### PR TITLE
Feature/te 377 add proxy config support

### PR DESCRIPTION
**Pull request by kpanier**
Config for Proxy in Eclipse, Java, and maven.

### CODE REVIEW

File: common/org.testeditor.dsl.common.ui/src/org/testeditor/dsl/common/ui/utils/ProjectContentGenerator.xtend [hunk 1]
```diff
@@ -145,6 +145,10 @@ class ProjectContentGenerator {
 	protected def void setupMavenProject(IProject project, String[] fixtures, IProgressMonitor monitor) {
 		var IFile buildFile = project.getFile("pom.xml")
 		buildFile.create(new StringInputStream(getPomContent(fixtures, project.name)), IResource.NONE, monitor)
+		val mavenSettings = System.getProperty("TE.MAVENSETTINGSPATH", "")
+		if (mavenSettings.length > 0) {
```
Comment (franzbecker) [inline common/org.testeditor.dsl.common.ui/src/org/testeditor/dsl/common/ui/utils/ProjectContentGenerator.xtend]: In Xtend you can use `mavenSettings.isNullOrEmpty`. Then there is no need to return an empty default String above.

File: rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java [hunk 2]
```diff
@@ -49,7 +51,16 @@
 	public int execute(String parameters, String pathToPom) {
 		System.setProperty("maven.multiModuleProjectDirectory", pathToPom);
 		MavenCli cli = new MavenCli();
-		int result = cli.doMain(parameters.split(" "), pathToPom, System.out, System.err);
+		List<String> params = new ArrayList<String>();
+		params.addAll(Arrays.asList(parameters.split(" ")));
+		String mavenSettings = System.getProperty("TE.MAVENSETTINGSPATH");
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java]: We got the retrieval (and the constant) twice here. Maybe we can extract this code into a helper class?

File: rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java [hunk 3]
```diff
@@ -75,15 +86,27 @@ public int executeInNewJvm(String parameters, String pathToPom, String testParam
 		command.add(jvm);
 		command.add("-cp");
 		command.add(getClassPath());
+		Properties props = System.getProperties();
+		for (String key : props.stringPropertyNames()) {
+			if (key.startsWith("http.")) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java]: What about the `https` proxy settings?

File: rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java [hunk 4]
```diff
@@ -75,15 +86,27 @@ public int executeInNewJvm(String parameters, String pathToPom, String testParam
...
+				command.add("-D" + key + "=" + props.getProperty(key));
+			}
+		}
+		String mavenSettings = System.getProperty("TE.MAVENSETTINGSPATH");
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java]: The constant is repeated here again.

File: rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java [hunk 5]
```diff
@@ -49,7 +51,16 @@
 	public int execute(String parameters, String pathToPom) {
 		System.setProperty("maven.multiModuleProjectDirectory", pathToPom);
 		MavenCli cli = new MavenCli();
-		int result = cli.doMain(parameters.split(" "), pathToPom, System.out, System.err);
+		List<String> params = new ArrayList<String>();
+		params.addAll(Arrays.asList(parameters.split(" ")));
+		String mavenSettings = System.getProperty("TE.MAVENSETTINGSPATH");
+		if (mavenSettings != null) {
+			if (mavenSettings.length() > 0) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java]: We can add a static import to `org.eclipse.xtext.xbase.lib.StringExtensions.isNullOrEmpty(String)` here to make the code more readable.

File: rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java [hunk 6]
```diff
@@ -75,15 +86,27 @@ public int executeInNewJvm(String parameters, String pathToPom, String testParam
...
 		command.add(this.getClass().getName());
 		command.add(parameters);
 		command.add(pathToPom);
 		command.add(testParam);
-		logger.trace("Execute maven in new jvm with: {}", command);
 		ProcessBuilder processBuilder = new ProcessBuilder();
 		processBuilder.inheritIO();
 		processBuilder.directory(new File(pathToPom));
 		processBuilder.redirectErrorStream(true);
+		logger.info("Execute maven in new jvm with: {}", command);
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java]: I would prefer "Executing". Also our logging conventions propose to write the statement like this: `... in new jvm with command='{}'."`

File: rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+						updateNetworkSateWidget("work offline", SWT.COLOR_DARK_YELLOW);
+					} else {
+						updateNetworkSateWidget("connected", SWT.COLOR_DARK_GREEN);
+					}
+				} else {
+					updateNetworkSateWidget("diconnected", SWT.COLOR_RED);
+				}
+			}
+		};
+		new Thread(runnable).start();
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java]: Why do we need to run this in a separate thread? The called method calls `Display.asyncExec` anyway.

File: rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+	}
+
+	protected void updateNetworkSateWidget(final String message, final int color) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java]: Typo here, should be `updateNetworkStateWidget`.

File: rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+	@Override
+	protected Point getInitialSize() {
+		return new Point(500, 380);
+	}
+
+	@Override
+	protected void okPressed() {
+		prefs.putBoolean(WORKOFFLINE, workOffline);
+		prefs.put(PATH_TO_MAVENSETTINGS, pathToMavenSettingsFileSettings);
+		prefs.put(PROXY_HOST, proxyHostSetting);
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java]: I'm missing the `https` settings here.

Comment (kpanier) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java]: Ok I added the same config values to the https variables. To keep the ui easy, i prefer to have one proxy config until somebody needs a seond one.

Comment (franzbecker) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java]: Yes, definitely. I would also consider this an unlikely case and if so the user has to go to the real preferences page and not our simplified dialog.

File: rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java [hunk 10]
```diff
@@ -106,7 +98,7 @@ public int executeInNewJvm(String parameters, String pathToPom, String testParam
 		processBuilder.inheritIO();
 		processBuilder.directory(new File(pathToPom));
 		processBuilder.redirectErrorStream(true);
-		logger.info("Execute maven in new jvm with: {}", command);
+		logger.info("Executing maven in new jvm with command=	{}", command);
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java]: There's a tab behind the `=`. Was this intentional?

File: rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java [hunk 11]
```diff
@@ -208,30 +210,24 @@ public void widgetSelected(SelectionEvent e) {
...
-		new Thread(runnable).start();
-	}
 
-	protected void updateNetworkSateWidget(final String message, final int color) {
-		Display.getDefault().asyncExec(new Runnable() {
+			}
 
-			@Override
-			public void run() {
+			public void upDateUI(String message, int color) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4/src/org/testeditor/rcp4/dialogs/NetworkConnectionSettingDialog.java]: Should be called `updateUI` to be consistent with other method names.

**Pull request merged**